src/auth/strategies/demoStrategy.js

```diff
@@ -0,0 +1,27 @@
+const { HttpStatusError } = require('common-errors');
+
+function verifyToken(token) {
+  const validToken = !!token; // check token structure, check token on exist
+
+  if (!validToken) {
+    throw new HttpStatusError(403, 'Invalid Token');
+  }
+}
+
+function demoStrategy(request) {
+  const { action } = request;
+  const { auth } = action;
+  const { strategy = 'required' } = auth;
+  const { authorization } = request.headers;
+
+  if (authorization) {
+    const [auth, token] = authorization.split(/\s+/, 2); // Authorization: Bearer [token]
+    return verifyToken(token)
+  }
+
+  if (strategy === 'required') {
```

Comment:
It's faster to check whether the token is missing or not, and then verify it. Do not check it once again on the `verifyToken` step. Since it's a demo example you might want to make demo check instead like `Authorization: Bearer demo` is ok, others are not, so it will be clear that `verifyToken()` is intended to verify the token content.